plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMemoryCacheFileOperations.java

```diff
@@ -0,0 +1,159 @@
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultiset;
+import com.google.common.collect.Multiset;
+import io.opentelemetry.api.common.Attributes;
+import io.trino.plugin.iceberg.util.FileOperationUtils;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.filesystem.tracing.CacheSystemAttributes.CACHE_FILE_LOCATION;
+import static io.trino.plugin.iceberg.IcebergQueryRunner.ICEBERG_CATALOG;
+import static io.trino.plugin.iceberg.util.FileOperationUtils.FileType.DATA;
+import static io.trino.plugin.iceberg.util.FileOperationUtils.FileType.MANIFEST;
+import static io.trino.plugin.iceberg.util.FileOperationUtils.FileType.METADATA_JSON;
+import static io.trino.plugin.iceberg.util.FileOperationUtils.FileType.SNAPSHOT;
+import static io.trino.testing.MultisetAssertions.assertMultisetsEqual;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toCollection;
+
+@Execution(ExecutionMode.SAME_THREAD)
+public class TestIcebergMemoryCacheFileOperations
+        extends AbstractTestQueryFramework
+{
+    public static final String TEST_SCHEMA = "test_memory_schema";
+    private Path cacheDirectory;
+
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        cacheDirectory = Files.createTempDirectory("cache");
+        closeAfterClass(() -> deleteRecursively(cacheDirectory, ALLOW_INSECURE));
+        Path metastoreDirectory = Files.createTempDirectory(ICEBERG_CATALOG);
+        closeAfterClass(() -> deleteRecursively(metastoreDirectory, ALLOW_INSECURE));
+
+        Map<String, String> icebergProperties = ImmutableMap.<String, String>builder()
+                .put("fs.coordinator-small-files-cache.enabled", "true")
+                .put("hive.metastore.catalog.dir", metastoreDirectory.toUri().toString())
+                .buildOrThrow();
+
+        DistributedQueryRunner queryRunner = IcebergQueryRunner.builder()
+                .setSchemaInitializer(SchemaInitializer.builder()
+                        .withSchemaName(TEST_SCHEMA)
+                        .build())
+                .setIcebergProperties(icebergProperties)
+                .setWorkerCount(0)
+                .build();
+        queryRunner.execute("CREATE SCHEMA IF NOT EXISTS " + TEST_SCHEMA);
+        return queryRunner;
+    }
+
+    @Test
+    public void testCacheFileOperations()
```

Comment:
We might want to trace the getLength calls which go to object storage as well.

Reply:
added